fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -155,21 +155,26 @@ def _call_oracle(self, lambda_vec):
 
         redY_unique = np.unique(redY)
 
-        classifier = None
+        estimator = None
         if len(redY_unique) == 1:
             logger.debug("redY had single value. Using DummyClassifier")
-            classifier = DummyClassifier(strategy='constant',
-                                         constant=redY_unique[0])
+            estimator = DummyClassifier(strategy='constant',
+                                        constant=redY_unique[0])
             self.n_oracle_calls_dummy_returned += 1
         else:
-            classifier = pickle.loads(self.pickled_estimator)
+            # use sklearn.base.clone to clone the estimator.
+            # It is the same as copy.deepcopy for non-sklearn estimators (By using safe=False).
+            # For sklearn estimators, it is more efficient as it would not
+            # do a copy.deepcopy. Rather, instantiate a new estimator using the
+            # get_params() internally.
+            estimator = clone(estimator=self.estimator, safe=False)
 
         oracle_call_start_time = time()
-        classifier.fit(self.X, redY, sample_weight=redW)
+        estimator.fit(self.X, redY, sample_weight=redW)
         self.oracle_execution_times.append(time() - oracle_call_start_time)
         self.n_oracle_calls += 1
 
-        return classifier
+        return estimator
 
     def best_h(self, lambda_vec):
```

Comment:
should this (and other methods) also call `check_is_fitted`? Or do we rely on the underlying estimator to do that?

Reply:
I think the `_call_oracle` method calls `fit` here. Are you referring to invoking the `check_is_fitted` before that? I think we rely on the underlying estimator for this AFAIK. @romanlutz @riedgar-ms would know more.

Reply:
@adrinjalali we're calling fit right before that. Any particular reason to call the check method here? I'm probably missing something.

Reply:
yeah, sorry, my bad. Ignore this one :D